src/topology/metric_space/basic.lean

```diff
@@ -672,6 +672,28 @@ lemma is_open_singleton_iff {X : Type*} [metric_space X] {x : X} :
   is_open ({x} : set X) ↔ ∃ ε > 0, ∀ y, dist y x < ε → y = x :=
 by simp [is_open_iff, subset_singleton_iff, mem_ball]
 
+/-- Given a point `x` in a discrete subset `s` of a metric space, there is an open ball
+centered at `x` and intersecting `s` only at `x`. -/
+lemma ball_of_mem_discrete [discrete_topology s] {x : α} (hx : x ∈ s) :
+∃ ε > 0, ∀ y ∈ metric.ball x ε, y ≠ x → y ∉ s :=
```

Comment:
Normally, this should be indented by two spaces (like every line in a theorem statement except the first one).

Also, wouldn't the statement be more natural in the form `∃ ε > 0, metric.ball x ε ∩ s = {x}`, just like you do for `nhd_singleton_of_mem_discrete`? For the name, what about `exists_ball_inter_eq_singleton_of_discrete` and for the other one `exists_nhd_inter_eq_singleton_of_discrete`?

Reply:
I corrected the indentation issue, here and in the subsequent lemma, thanks!

As for the formulation, this lemma was driven by the proof of another one, where the expected term was the one appearing in this statement.  Before I change this lemma to the different formulation, I will change the intended application, to see that nothing breaks!

Reply:
I modified the two lemmas and changed their names.  I opted for `exists_closed_ball_inter_eq_singleton_of_discrete` instead of `exists_nhd_inter_eq_singleton_of_discrete`, since, for instance in my intended application, it was important that it was a *closed* neighbourhood.  Otherwise, I guess that the other lemma was also about a `nhd` and I thought that I could get confused!

Anyway, thanks for your suggestions!  The proofs are not as slick as the ones that Patrick had given, but they work!